storm-core/test/jvm/org/apache/storm/serialization/SerializationTest.java

```diff
@@ -0,0 +1,160 @@
+        return serializer.serialize(vals);
+    }
+
+    public List deserialize(byte[] bytes, Map conf) throws IOException {
+        KryoValuesDeserializer deserializer = new KryoValuesDeserializer(mkConf(conf));
+        return deserializer.deserialize(bytes);
+    }
+
+    public List roundtrip(List vals) {
+        return roundtrip(vals, new HashMap());
+    }
+
+    public List roundtrip(List vals, Map conf) {
+        List ret = null;
+        try {
+            ret = deserialize(serialize(vals, conf), conf);
+        } catch (IOException e) {
+            LOG.error("Exception when serialize/deserialize ", e);
+        }
+        return ret;
+    }
+
+    @Test
+    public void testJavaSerialization() {
+        Object obj = new TestSerObject(1, 2);
+        List<Object> vals = Lists.newArrayList(obj);
+
+        Map<String, Object> conf = new HashMap<>();
+        conf.put(Config.TOPOLOGY_KRYO_REGISTER, new HashMap<String, String>() {{
+            put("org.apache.storm.testing.TestSerObject", null);
+        }});
+        conf.put(Config.TOPOLOGY_FALL_BACK_ON_JAVA_SERIALIZATION, false);
+        try {
+            roundtrip(vals, conf);
+            Assert.fail("Expected Exception not Thrown for config: " + conf);
+        } catch (Exception e) {
+        }
+
+        conf.clear();
+        conf.put(Config.TOPOLOGY_FALL_BACK_ON_JAVA_SERIALIZATION, true);
+        Assert.assertEquals(vals, roundtrip(vals, conf));
+    }
+
+    @Test
+    public void testKryoDecorator() {
+        Object obj = new TestSerObject(1, 2);
+        List<Object> vals = Lists.newArrayList(obj);
+
+        Map<String, Object> conf = new HashMap<>();
+        conf.put(Config.TOPOLOGY_FALL_BACK_ON_JAVA_SERIALIZATION, false);
+        try {
+            roundtrip(vals, conf);
+            Assert.fail("Expected Exception not Thrown for config: " + conf);
+        } catch (Exception e) {
+        }
+
+        conf.put(Config.TOPOLOGY_KRYO_DECORATORS, Lists.newArrayList("org.apache.storm.testing.TestKryoDecorator"));
+        Assert.assertEquals(vals, roundtrip(vals, conf));
+    }
+
+
+    public static String mkString(int size) {
+        StringBuilder sb = new StringBuilder();
+        while (size-- > 0) {
+            sb.append("a");
+        }
+        return sb.toString();
+    }
+
+    public void isRoundtrip(List vals) {
+        Assert.assertEquals(vals, roundtrip(vals));
+    }
+
+    @Test
+    public void testStringSerialization() {
+        isRoundtrip(Lists.newArrayList("a", "bb", "cbe"));
+        isRoundtrip(Lists.newArrayList(mkString(64 * 1024)));
+        isRoundtrip(Lists.newArrayList(mkString(1024 * 1024)));
+        isRoundtrip(Lists.newArrayList(mkString(1024 * 1024 * 2)));
+    }
+
+    /** when we move to java totally, we can remove this test function entirely **/
+    @Test
+    public void testClojureSerialization() {
+        isRoundtrip(PersistentVector.create(RT.keyword(null, "a")));
```

Comment:
I tried to leave `test-clojure-serialization` in the clojure file, but it depends on many util functions such as: `is-roundtrip` `serialize` `deserialize`.  I can't call isRoundtrip in clojure even if I set isRoundtrip `public static`, as `SerializationTest` is a Test Class.  If I leave the origin `is-roundtrip` `serialize` `deserialize` functions in clojure, there will be a duplication of code, which is very odd.

So I still translate `test-clojure-serialization` to java and add a TODO comment on it . When we move to java completely , we can remove this test function entirly.

If you have any ideas, please let me know.